broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -601,18 +605,22 @@ public void onRecovered() {
   private void updateHealthStatus(final HealthStatus newStatus) {
     if (healthStatus != newStatus) {
       healthStatus = newStatus;
-      if (failureListener != null) {
-        switch (newStatus) {
-          case HEALTHY:
+      switch (newStatus) {
+        case HEALTHY:
+          healthMetrics.setHealthy();
+          if (failureListener != null) {
             failureListener.onRecovered();
-            break;
-          case UNHEALTHY:
+          }
+          break;
+        case UNHEALTHY:
+          healthMetrics.setUnhealthy();
+          if (failureListener != null) {
             failureListener.onFailure();
```

Comment:
Can this call fail? E.g. Exception?

Reply:
Probably not. But in case it fails, what would be the expected behavior? If it is just for logging, it will be logged by the actor controller. Health monitor is also periodically probing these modules. So it will detect the failure of this module eventually.

Reply:
> So it will detect the failure of this module eventually.

Ok here I was unsure. I thought that we might miss an failure. E. g. we report healthy and the failureListener throws an exception that the actors might then have a problem but we are not seeing it?

Reply:
If just the notification to failure listener fails, health monitor will detect the health status correctly in the next probe. Do you suggest to handle it any other way?

Reply:
How often the probe is done? Sorry I think I have not the complete picture currently.

Reply:
Currently hard coded to 60 seconds.

Reply:
Ok thanks yes should be fine